std/internal/cstring.d

```diff
@@ -91,41 +101,86 @@ if(isSomeChar!To && isSomeChar!From)
         @disable this(this);
         alias ptr this;
 
-        @property inout(To)* buffPtr() inout @safe pure
-        { return _ptr == useStack ? _buff.ptr : _ptr; }
+        @property inout(To)* buffPtr() inout pure
+        {
+            return _ptr == useStack ? _buff.ptr : _ptr;
+        }
 
-        @property const(To)* ptr() const @safe pure
-        { return buffPtr; }
+        @property const(To)* ptr() const pure
+        {
+            return buffPtr;
+        }
 
         ~this()
-        { if(_ptr != useStack) rawFree(_ptr); }
+        {
+            if (_ptr != useStack)
+            {
+                import core.stdc.stdlib : free;
+                free(_ptr);
+            }
+        }
 
     private:
         To* _ptr;
-        To[256] _buff;
-    }
-
-    // TODO: Don't stack allocate uninitialized array to
-    // not confuse unprecise GC.
+        version (unittest)
+        {
+            enum buffLength = 16 / To.sizeof;   // smaller size to trigger reallocations
```

Comment:
Yeah, the private enum could be inside the function instead of the type, but I think there's no risk of this leaking outside the file somehow. So it works, thanks.